machine/quanta/quanta_common_p2020/u-boot/platform-quanta-common_p2020.patch

```diff
@@ -0,0 +1,2235 @@
++#endif
++
++/*
++ * I2C2 EEPROM
++ */
++#define CONFIG_CMD_EEPROM		/* EEPROM read/write support	*/
++
++#define CONFIG_CMD_SYS_EEPROM
++
++/*
++ * bit of hack to enable reading MAC from EEPROM w/o using Freescale's
++ * 'mac' command.  Just having it defined enables the infra to call
++ * our mac_read_from_eeprom() to set $ethaddr.
++ *
++ */
++#define CONFIG_SYS_I2C_MAC_OFFSET
++#define CONFIG_SYS_EEPROM_BUS_NUM		0
++#define CONFIG_SYS_EEPROM_PAGE_WRITE_DELAY_MS	5
++#define CONFIG_SYS_I2C_EEPROM_ADDR		0x54
++#define CONFIG_SYS_I2C_EEPROM_ADDR_LEN		1
++#define CONFIG_SYS_I2C_EEPROM_INFO_ADDR         0x0
++#define CONFIG_SYS_I2C_EEPROM_INFO_LEN          0x0
++#define CONFIG_SYS_EEPROM_MAX_SIZE		256
++#define CONFIG_SYS_EEPROM_LOAD_ENV_MAC
++#define CONFIG_SYS_EEPROM_USE_COMMON_I2C_IO
++#define CONFIG_SYS_EEPROM_PAGE_WRITE_BITS	3
++#define CONFIG_SYS_EEPROM_MAX_NUM_ETH_PORTS	1
++
++#if 0
++/*
++ * Quanta Board Info
++ */
++
++#define CONFIG_QUANTA_SOFT_VERSION 0x0000fff1
++#define CONFIG_COMMON_P2020
++#define I2C_AT24C02_EEPROM_MB 0x54
++#define EEPROM_24C02_SIZE 256
++#define I2C_PCA9546_FAN_I2C_SWITCH 0x71
++#define MODEL_NAME CONFIG_QUANTA_MODEL_NAME
++#define CONFIG_QUANTA_MODEL_NAME "COMMON P2020"
++#endif
++
++#undef	CONFIG_WATCHDOG		/* watchdog disabled */
++
++/*
++ * Miscellaneous configurable options
++ */
++#define CONFIG_SYS_LOAD_ADDR	0x2000000	/* default load address */
++#define CONFIG_SYS_PROMPT	"=> "		/* Monitor Command Prompt */
++
++#if defined(CONFIG_CMD_KGDB)
++#define CONFIG_KGDB_BAUDRATE	230400	/* speed to run kgdb serial port */
++#define CONFIG_KGDB_SER_INDEX	2	/* which serial port to use */
++#endif
++
++/*
++ * Environment Configuration
++ */
++
++#if defined(CONFIG_TSEC_ENET)
++#define CONFIG_HAS_ETH0
++#endif
++
++#define CONFIG_HOSTNAME		quanta-common-p2020-unknown
++
++/*default location for tftp and bootm*/
++#define CONFIG_LOADADDR		CONFIG_SYS_LOAD_ADDR
++
++/*
++** Platform environment variables needed by the common env
++** infrastructure.
++**
++*/
++#define CONFIG_PLATFORM_ENV	\
++	"consoledev=ttyS0\0"	\
++	"onie_start=0xefb60000\0"	\
++	"onie_sz.b=0x00400000\0"
++
++#define	CONFIG_EXTRA_ENV_SETTINGS		\
++	CONFIG_PLATFORM_ENV			\
++	CONFIG_ONIE_COMMON_UBOOT_ENV
++
++#endif	/* __CONFIG_H */
+diff --git a/include/configs/onie_common_config.h b/include/configs/onie_common_config.h
+index c941e2a..5dbfd7f 100644
+--- a/include/configs/onie_common_config.h
++++ b/include/configs/onie_common_config.h
+@@ -35,6 +35,16 @@
+ #define CONFIG_FDT_ENV_SIZE
+ 
+ /*
++ * onie_preboot changes fdt compatible and model to quanta,${motherboard_platform}
++ */
++#define CONFIG_ONIE_PREBOOT \
++        "dtbaddr=0x8000000\0"   \
++        "onie_preboot=cp.b $onie_start $loadaddr ${onie_sz.b} && "      \
++        "imxtract $loadaddr dtb $dtbaddr ; fdt addr $dtbaddr ; "        \
++        "fdt set / compatible quanta,${motherboard_platform} ; "        \
++        "fdt set / model quanta,${motherboard_platform}\0"
```

Comment:
What happens if `imxtract` fails?  All the remaining command will still be executed.

It might be better to use `&&` between all the commands instead of `;`.